### ISSUE TITLE
Ability to specify multiple values for a single key in `UploadTask` and `MultiUploadTask` form-data

### ISSUE THREAD

**Issue by Vanchel** [781flyingdutchman/background_downloader#245]
Hello. Is it possible to specify multiple values for a single key in the `fields` parameter when using `UploadTask` (and `MultiUploadTask`)? An example can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append).

I have reviewed the documentation and API but have not been able to find a solution.

Am I correct in understanding that when using the `files` field in `MultiUploadTask`, such behavior can be achieved using the following structure:
`files: [('files[]', image1.path), ('files[]', image2.path)]`?

If not, is there a way to implement this behavior? Additionally, is it possible to achieve the same for the `fields` parameter? If not, it would be great if this capability could be added.

**Comment by 781flyingdutchman** [comment]
Hi, thanks for bringing this to my attention. You are correct that this is currently possible only for the `files` field, as your example demonstrates. It is not currently possible for the `fields` parameter but you're in luck because I had some time and implemented this on branch multi_fields. Change your pubspec.yaml to:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: multi_fields
```

New docs: For multi-part uploads you can specify name/value pairs in the `fields` property of the `UploadTask` as a `Map<String, String>`. These will be uploaded as form fields along with the file. To specify multiple values for a single name, format the value as `'"value1", "value2", "value3"'` (note the double quotes and the comma to separate the values).

Can you try out if this works for you and let me know? If it works, I'll push it as a new version.

**Comment by Vanchel** [comment]
I've tried it, everything works. Thank you very much, especially for your quick response.

**Comment by 781flyingdutchman** [comment]
Great!  Published as V8.2.1
